taglib/fileref.cpp

```diff
@@ -84,6 +84,23 @@ namespace
     return 0;
   }
 
+  File *detectByResolvers(IOStream* stream, bool readAudioProperties,
+                          AudioProperties::ReadStyle audioPropertiesStyle)
+  {
+    ResolverList::ConstIterator it = fileTypeResolvers.begin();
+    for(; it != fileTypeResolvers.end(); ++it) {
+      const FileRef::StreamTypeResolver* streamResolver =
+        dynamic_cast<const FileRef::StreamTypeResolver*>(*it);
+      if (!streamResolver)
+        continue;
+      File *file = streamResolver->createFileFromStream(stream, readAudioProperties, audioPropertiesStyle);
+      if(file)
+        return file;
+    }
+
+    return 0;
```

Comment:
```suggestion
  File *detectByResolvers(IOStream* stream, bool readAudioProperties,
                          AudioProperties::ReadStyle audioPropertiesStyle)
  {
    for(ResolverList::ConstIterator it = fileTypeResolvers.begin();
        it != fileTypeResolvers.end(); ++it) {
      if(const FileRef::StreamTypeResolver *streamResolver =
           dynamic_cast<const FileRef::StreamTypeResolver*>(*it)) {
        if(File *file = streamResolver->createFileFromStream(
             stream, readAudioProperties, audioPropertiesStyle)) {
          return file;
        }
      }
    }

    return 0;
```
I think it is better to narrow the scope of variables and only use `continue` when it would result in more readable code.

Reply:
I'm fine with that change, but if I'm not mistaken this would require C++17 while taglib only requires C++11

Reply:
I don't think TagLib actually requires C++11.  At least in the not terribly recent past, it was all still C++98, and I'd removed the couple places that had C++11-isms.